torch_geometric/datasets/metrla.py

```diff
@@ -0,0 +1,177 @@
+            the features.
+        add_day_of_week (bool): Whether to inject time of day information in
+            the features.
+        normalized_k (float):  The threshold for constructing the adjacency
+            matrix based on the thresholded Gaussian kernel.
+        transform: A function/transform that takes in an
+            :obj:`torch_geometric.data.Data` object and returns a transformed
+            version. The data object will be transformed before every access.
+        pre_transform: A function/transform that takes in an
+            :obj:`torch_geometric.data.Data` object and returns a transformed
+            version. The data object will be transformed before being saved to
+            disk
+    """
+    gdrive_ids = {
+        'distances.csv': '19Td6JafGnF8CD2H64jWCBV7k5GyFix2H',
+        'locations.csv': '1FnioVF2jZuOl_St1ssLnvQgSHuEuDzvL',
+        'sensors.txt': '1235bXBxe4X73dJk3zwRzaIQ-8A7K_OEV',
+        'sensor-readings.h5': '1VgLjs4XB5O-IpcyZ5GAQG9sGuyxKe3Lc'
+    }
+
+    def __init__(self, root: Optional[str], n_previous_steps: int,
+                 n_future_steps: int, add_time_of_day: bool = False,
+                 add_day_of_week: bool = False, normalized_k: float = .1,
+                 transform: Optional[Callable] = None,
+                 pre_transform: Optional[Callable] = None) -> None:
+        """This is a circular dependency. You need this to calculate processed
+        file names, but you need it before the call call to the "process"
+        method in the superclass. The only in-between place you could
+        effectively calculate this is before the process, so either in
+        __init__ (where you don't have access to the downloaded file
+        containing the readings, or in download, but download gets skipped
+        after the first time, so that's not reliable. Open to any
+        workarounds, but the dataset has a fixed size, so
+        this is not a deal breaker imo."""
+        self.n_readings = 34272
+
+        self.io = MetrLaIo(n_readings=self.n_readings,
+                           n_previous_steps=n_previous_steps,
+                           n_future_steps=n_future_steps,
+                           add_time_of_day=add_time_of_day,
+                           add_day_of_week=add_day_of_week,
+                           normalized_k=normalized_k)
+        super().__init__(root, transform, pre_transform)
+
+        self.normalized_k = normalized_k
+
+    @property
+    def raw_file_names(self) -> str:
+        r"""The name of the files in the :obj:`self.raw_dir`
+        folder that must be present in order to skip downloading."""
+        return list(self.gdrive_ids.keys())
+
+    @property
+    def processed_file_names(self) -> str:
+        r"""The name of the files in the :obj:`self.processed_dir`
+        folder that must be present in order to skip processing."""
+        return [f'data_{i}.pt' for i in range(self.io.dataset_len)
```

Comment:
```suggestion
        return [f'data_{i}.pt' for i in range(5)
```

Reply:
Right, but then the problem is in the "get" function. But I guess I can solve that with an f-string (assuming all the files are named with the same template). Is that an appropriate way to go?

Reply:
I wrapped the getter of the name in a private method (as it was used more than twice, to avoid further problems with renaming, and to avoid possible typos with hard-coded values throughout the code).